### PR TITLE
Update to work with latest WebIDLs and enable WebGL2

### PR DESCRIPTION
**Pull request by haxiomic**
- Update to latest WebIDLs and WebIDL.py from [mozilla/gecko-dev](http://github.com/mozilla/gecko-dev)
- Enable generating WebGL2 externs in `bin/haxe.py`
- Applied fixes to `bin/generate` and `bin/haxe.py` to enable supporting the latest WebIDLs

(Related haxe issue https://github.com/HaxeFoundation/haxe/issues/5770)

Remaining todo before master PR:
- [x] Pass validation
- [x] Add missing fields from TypedArray externs
- [x] Better auto-documentation for WebGL externs
- [x] Manually verify all changes to generated externs
- [x] Deprecation / migration list
- [x] Test popular projects to ensure there's no breaking changes

Generated externs diff: https://github.com/HaxeFoundation/haxe/pull/7354

### CODE REVIEW

File: bin/Haxe.py [hunk 1]
```diff
@@ -735,6 +747,8 @@ def toHaxeType (name):
 	name = re.sub("^HTML(.+Element)", "\\1", name)
 	if name.startswith("SVG"):
 		name = name[len("SVG"):]
+	elif name.startswith("WebGL2"):
```
Comment (haxiomic) [inline bin/Haxe.py]: This change means we generate `webgl/RenderingContext.hx` and `webgl/WebGL2RenderingContext.hx`, which I thought was nicer than `RenderingContext2.hx`. Unfortunately I don't think it's possible to split the definitions up into a separate `webgl2` directory

**Comment by kLabz** [comment]
Wow, nice!
Does it fix #6 ?

**Comment by haxiomic** [comment]
Hey @kLabz thanks for pointing #6 that out, it turns out that field has to be explicitly enabled which I've done in 891b31f

**Comment by nadako** [comment]
I have no idea how to review this properly, but you do seem to know what are you doing and I love the commit messages :) I guess we should try merging this and then trying to build popular projects (openfl, kha, heaps, etc.) that are using WebGL

**Comment by Simn** [comment]
I agree. I'll add you as a contributor too.

**Comment by Simn** [comment]
(Feel free to merge this whenever you think it's ready!)

**Comment by haxiomic** [comment]
Thanks @Simn and @nadako :), I've added a task to test popular projects and I'll merge after I've confirmed nothing fails

**Comment by haxiomic** [comment]
Hey @markknol,

I found I was changing the doc script quite frequently, usually to sync up with changes in the generator. To make the externs easier to reproduce by others it seemed sensible to bring all the scripts to the same place (previously someone would need to find my fork for docs). That way it would also be easier to match a given version of externs with a given version of the generator scripts

However happy to revert this and PR to the original repo since it's not something that affects my workflow

**Comment by markknol** [comment]
No it's fine, what ever fits the needs. I think in that case Ill archive/deprecate my repository. 
I'm already happy you got this externs update machine working 🥇

**Pull request merged**